lib-ee/emqx_license/src/emqx_license_parser.erl

```diff
@@ -0,0 +1,139 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2022 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% @doc EMQ X License Management.
+%%--------------------------------------------------------------------
+
+-module(emqx_license_parser).
+
+-include_lib("emqx/include/logger.hrl").
+-include("emqx_license.hrl").
+
+% -define(PUBKEY, <<"MIIBCgKCAQEAyJgH+BvEJIStYvyw1keQ/ixVPJ4GGjlP7lTKnZL
+%                   3mqZyPXQUEaLnRmcQ3/ra8xYQPcfMReynqmrYmT45/eD2tK7rdXT
+%                   zYfOWoU0cXNQMaQS7be1bLF4QrAEbJhEsgkjX9rP30mrzZCjRRnk
+%                   QtWmi4DNBU4qOl6Ee9aAS5aY+L7DW646J47Tyc7gAA4sdZw04KGB
+%                   XnSyXzyBvPaf+QByOrOXUxBcxehHN/Ox41/duYFXSR40U6lyp49N
+%                   YJ6yEUVWSp4oxsrkcgqyegNKXdW1D8oqJXzxalbh/RB8YYlX+Ae3
+%                   77gEBlLefPFdSEYDRN/ajs3UIeqde6i20lVyDPIjEcQIDAQAB">>).
+
+
+-define(PUBKEY, <<"MEgCQQChzN6lCUdt4sYPQmWBYA3b8Zk87Jfk+1A1zcTd+lCU0Tf
+                  vXhSHgEWz18No4lL2v1n+70CoYpc2fzfhNJitgnV9AgMBAAE=">>).
+
+% -define(DIGEST, <<"3jHg0zCb4NL5v8eIoKn+CNDMq8A04mXEOefqlUBSSVs=">>).
+
+-define(DIGEST, <<"iakqYnKDoLzHj9sbnVJD/N6iFrsGyBZpvLwpS0Rvots=">>).
+
+-define(LICENSE_PARSE_MODULES, [emqx_license_parser_v20220101
+                               ]).
+
+-type license_data() :: term().
+-type customer_type() :: ?SMALL_CUSTOMER |
+                         ?MEDIUM_CUSTOMER |
+                         ?LARGE_CUSTOMER |
+                         ?EVALUATION_CUSTOMER.
+
+-type license_type() :: ?OFFICIAL | ?TRIAL.
+
+-type license() :: #{module := module(), data := license_data()}.
+
+-export_type([license_data/0,
+              customer_type/0,
+              license_type/0,
+              license/0]).
+
+
+-export([parse/1,
+         parse/2,
+         dump/1,
+         customer_type/1,
+         license_type/1,
+         expiry_date/1,
+         max_connections/1
+        ]).
+
+-ifdef(TEST).
+-export([public_key/0,
+         digest/0
+        ]).
+-endif.
+
+%%--------------------------------------------------------------------
+%% Behaviour
+%%--------------------------------------------------------------------
+
+-callback parse(string() | binary(), binary()) -> {ok, license_data()} | {error, term()}.
+
+-callback dump(license_data()) -> list({atom(), term()}).
+
+-callback customer_type(license_data()) -> customer_type().
+
+-callback license_type(license_data()) -> license_type().
+
+-callback expiry_date(license_data()) -> calendar:date().
+
+-callback max_connections(license_data()) -> non_neg_integer().
+
+%%--------------------------------------------------------------------
+%% API
+%%--------------------------------------------------------------------
+
+-spec parse(string() | binary()) -> {ok, license()} | {error, term()}.
+parse(Content) ->
+    DecodedKey = base64:decode(public_key()),
+    case verify_public_key(DecodedKey, digest()) of
```

Comment:
Since both `public_key/0` and `digest/0` are known at compile time, is this needed?

Maybe the intention was to verify the signature that would come associated with the contents?

Reply:
This is the inherited behaviour also.
I removed this digest verification.